packages/storage/amplify_storage_s3/example/integration_test/download_data_test.dart

```diff
@@ -0,0 +1,261 @@
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(jsonPath),
+        ).result;
+
+        expect(downloadResult.bytes, jsonEncode(json).toString().codeUnits);
+      });
+
+      test('formFields', () async {
+        const key = 'foo';
+        const value = 'bar';
+
+        final formFieldsPath = 'public/download-data-form-fields-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(formFieldsPath),
+          data: HttpPayload.formFields({key: value}),
+        ).result;
+
+        addTearDown(
+          () => Amplify.Storage.remove(
+            path: StoragePath.fromString(formFieldsPath),
+          ),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(formFieldsPath),
+        ).result;
+
+        expect(downloadResult.bytes, '$key=$value'.codeUnits);
+      });
+
+      test('string', () async {
+        const string = 'test string';
+
+        final stringPath = 'public/download-data-string-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(stringPath),
+          data: HttpPayload.string(string),
+        ).result;
+
+        addTearDown(
+          () =>
+              Amplify.Storage.remove(path: StoragePath.fromString(stringPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(stringPath),
+        ).result;
+
+        expect(downloadResult.bytes, string.codeUnits);
+      });
+
+      test('empty', () async {
+        final emptyPath = 'public/download-data-empty-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(emptyPath),
+          data: const HttpPayload.empty(),
+        ).result;
+
+        addTearDown(
+          () => Amplify.Storage.remove(path: StoragePath.fromString(emptyPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(emptyPath),
+        ).result;
+
+        expect(downloadResult.bytes, <int>[]);
+      });
+
+      test('stream', () async {
+        final data = [1, 2, 3];
+        final stream = Stream<List<int>>.value(data);
+
+        final streamPath = 'public/download-data-stream-${uuid()}';
+        await Amplify.Storage.uploadData(
+          path: StoragePath.fromString(streamPath),
+          data: HttpPayload.streaming(stream),
+        ).result;
+
+        addTearDown(
+          () =>
+              Amplify.Storage.remove(path: StoragePath.fromString(streamPath)),
+        );
+
+        final downloadResult = await Amplify.Storage.downloadData(
+          path: StoragePath.fromString(streamPath),
+        ).result;
+
+        expect(downloadResult.bytes, data);
+      });
+
+      test('with identity ID', () async {
```

Comment:
todo: move this out of the group it is in.

This shouldn't be grouped under "for data type". This isn't a testing a specific data type.